implants/eldritch/src/lib.rs

```diff
@@ -61,29 +61,38 @@ impl PrintHandler for StdPrintHandler {
 
 pub fn eldritch_run(tome_filename: String, tome_contents: String, tome_parameters: Option<String>, print_handler: &(dyn PrintHandler)) -> anyhow::Result<String> {
     // Boilder plate
-    let ast: AstModule;
-    match AstModule::parse(
+    let ast =  match AstModule::parse(
             &tome_filename,
             tome_contents.as_str().to_owned(),
             &Dialect::Standard
         ) {
-            Ok(res) => ast = res,
-            Err(err) => return Err(err),
-    }
+            Ok(res) => res,
+            Err(err) => return Err(anyhow::anyhow!("[eldritch] Unable to parse eldritch tome: {}: {} {}", err.to_string(), tome_filename.as_str(), tome_contents.as_str())),
+    };
 
     let tome_params_str: String = match tome_parameters {
-        Some(param_string) => param_string,
+        Some(local_param_string) => match local_param_string.as_str() {
+            "" => "{}".to_string(), // If we get "" as our params update it to "{}"
```

Comment:
why do we do this?

Reply:
The JSON param parser will fail out if we pass an empty string so need to pass an empty object.

An alternative would be to refactor "input_params" to be optional for the interpreter however I think it's better that we always have input_params exist in the eldritch tomes.
Without input_params errors where a user tries to use it will fail with a "input_params variable doesn't exist" whereas with an empty one it will fail with "{your_var} isn't a key in input_params" which IMO is more clear.